### PR TITLE
Fix failure when crossgen (*.ni.pdbs) and portable pdbs exists in the debug directory

### PR DESCRIPTION
**Pull request by mikem8361**
Start with the last debug directory entry instead of the first.

Also bumped the System.Reflection.Metadata version to 1.5.0 to work properly when running under msbuild on desktop. Otherwise it doesn't work for the DotNet-SymStore repo uploader build task.

### CODE REVIEW

File: src/Microsoft.DiaSymReader.Converter/Utilities/SymReaderHelpers.cs [hunk 1]
```diff
@@ -83,7 +83,7 @@ public static ImmutableArray<string> GetImportStrings(ISymUnmanagedReader reader
 
         public static bool TryReadPdbId(PEReader peReader, out BlobContentId id, out int age)
         {
-            var codeViewEntry = peReader.ReadDebugDirectory().FirstOrDefault(entry => entry.Type == DebugDirectoryEntryType.CodeView);
+            var codeViewEntry = peReader.ReadDebugDirectory().LastOrDefault(entry => entry.Type == DebugDirectoryEntryType.CodeView);
```
Comment (vancem) [inline src/Microsoft.DiaSymReader.Converter/Utilities/SymReaderHelpers.cs]: Mike for completeness there is another place you should update from FirstOrDefault to LastOrDefault

There is similar logic in GetPdbPathFromCodeViewEntry in the Pdb2Pdb.cs file (used when infering the PDB file name from the DLL).

**Comment by mikem8361** [comment]
Fixed it. Thanks.

**Comment by vancem** [comment]
Note that for others who may not be familiar with the semantics here.  The DebugDirectory is a list of PDBs that are associated with the DLL (it is how you find the PDB on the symbol server).   Most DLLs have only one entry, but .NET Native images have two (one for the IL code, one for the native code).   All debugger to date always choose the LAST entry (which is the IL entry).    This change is simply making the converter follow the same conventions as every other tool (if you only care about the IL PDB, choose the last entry, which will be the IL PDB).

**Comment by tmat** [comment]
Some feedback:
1) Could you add a small ngen'd hello world dll to PdbTestResources and add write a unit test for TryReadPdbId that validates the right record is found?
2) We should document what the semantics of multiple CodeView records is a spec: https://github.com/dotnet/corefx/blob/master/src/System.Reflection.Metadata/specs/PE-COFF.md
3) I think we should disallow multiple portable CodeView records. Is there a way to distinguish record for native PDB vs. record for managed PDB in the native image? If we can I'd also disallow multiple managed PDB records (or perhaps even multiple native PDB records). If we don't disallow multiple portable records we would need to fix https://github.com/dotnet/corefx/blob/master/src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEReader.cs#L693 to also pick the last one. This would also require update of .NET Framework, so I would rather not do that. There seems to be no point in allowing multiple portable records.

**Comment by mikem8361** [comment]
I’m not sure how to disallow multiple portable CodeView records. Crossgen adds the *.ni.pdb entry at the first one and just copies the rest.  The compiler (I think) would be the only one that would/could add multiple.

Do you mean validate and fail in the converter code if there is more than one portable pdb entry?

**Comment by tmat** [comment]
> Do you mean validate and fail in the converter code if there is more than one portable pdb entry?

Yes. The converter is also a verifier (think of PEVrify for PDBs). This doesn't need to be "fatal" error, we can go on and take the last entry, but it should report [diagnostic](https://github.com/dotnet/symreader-converter/blob/master/src/Microsoft.DiaSymReader.Converter/Diagnostics/PdbDiagnosticId.cs) that something's wrong with the PDB.

**Comment by mikem8361** [comment]
>>I think we should disallow multiple portable CodeView records. Is there a way to distinguish record for native PDB vs. record for managed PDB in the native image? If we can I'd also disallow multiple managed PDB records (or perhaps even multiple native PDB records).

There really isn’t a deterministic way to distinguish between native PDB and managed PDB debug entries other than the “.ni.” in the name.  We can only verify that there is only one portable pdb entry using the MinorVersion, but TryReadPdbId doesn’t look for portable pdb entries specifically and is used for converting in both directions so it can’t look for it.

I really don’t have a lot of time to do all the things you want. I’ll try to come up with a test and document this in the corefx repo.

**Comment by tmat** [comment]
> I really don’t have a lot of time to do all the things you want. 

That's fine. We don't need to block this PR on them, other then adding the test. Let's file issues for them and address them later.

**Comment by mikem8361** [comment]
One more question about uploading the result package(s).  When I create the packages with -pack they have a version of 1.1.0-dev (i.e. Microsoft.DiaSymReader.Converter.1.1.0-dev.nupkg) and the latest on the build tools feed is “1.1.0-beta1-62225-01".  What version should the new package be?

**Comment by tmat** [comment]
@mikem8361 You should not upload any nuget packages. They are produced and published by our Microbuild task.

**Comment by tmat** [comment]
👍

**Comment by tmat** [comment]
Thanks!

**Comment by mikem8361** [comment]
@tmat can you merge this PR? I don't have write access.

**Comment by tmat** [comment]
New official build will be published shortly on myget

**Comment by tmat** [comment]
Microsoft.DiaSymReader.Converter.1.1.0-beta1-62407-01.nupkg

**Comment by tmat** [comment]
Filed https://github.com/dotnet/symreader-converter/issues/105

**Pull request merged**